### PR TITLE
Fix BlockDirectiveParser logic

### LINKED ISSUE #49: BlockDirective range issue
For a multi-line block directive, it can parse properly.
eg.
```swift
import Markdown
let source = """
@Image(source: a.png, 
        alt: "hello")
"""
Document(parsing: source, options: [.parseBlockDirectives])
.debugDescription(options: .printSourceLocations)
```

```
Document @1:1-2:22 Root #1 #0
└─ BlockDirective @1:1-2:22 #1 name: "Image"
   ├─ Argument text segments:
   |    @1:8-1:23: "source: a.png, "
   |    @2:1-2:21: "        alt: \"hello\""
```

However, the single-line form behavior is not.

eg.
```swift
import Markdown
let source = """
@Image(source: a.png, alt: "hello")
"""
Document(parsing: source, options: [.parseBlockDirectives])
.debugDescription(options: .printSourceLocations)
```

## Actually behavior

```
Document @1:1-1:8 Root #1 #0
└─ BlockDirective @1:1-1:8 #1 name: "Image"
   ├─ Argument text segments:
   |    @1:8-1:35: "source: a.png, alt: \"hello\""
```

## Expected behavior

```
Document @1:1-1:36 Root #1 #0
└─ BlockDirective @1:1-1:36 #1 name: "Image"
   ├─ Argument text segments:
   |    @1:8-1:35: "source: a.png, alt: \"hello\""
```

## Other

This is effecting https://github.com/apple/swift-docc/pull/278

And a temporary workaround on swift-docc is 👇
```swift
// FIXME: - Bug on swift-markdown #49
guard let blockRange = blockDirective.range else { return }
let bound = blockDirective.argumentText.parseNameValueArguments().compactMap {
    $0.valueRange?.upperBound
}.max() ?? SourceLocation(line: 0, column: 0, source: blockRange.upperBound.source)
let upperBound = SourceLocation(line: bound.line, column: blockRange.upperBound.column + bound.column, source: bound.source)
let range = blockRange.lowerBound ..< upperBound
// let range = blockRange
```
> The workaround is buggy and will not work if the last argument's value is surrounded by quotes.

### PR DESCRIPTION
**Pull request by Kyle-Ye**
Bug/issue #, if applicable: 

## Summary

Close #49 

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran the `./bin/test` script and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
@swift-ci  please test

**Comment by QuietMisdreavus** [comment]
This looks nice to me, but i'll defer to @franklinsch for the final call.

**Comment by Kyle-Ye** [comment]
@swift-ci please test

File: Sources/Markdown/Parser/BlockDirectiveParser.swift [hunk 1]
```diff
@@ -101,8 +101,8 @@ struct PendingBlockDirective {
             // There may be garbage after the left parenthesis `(`, but we'll
             // still consider subsequent lines for argument text, so we'll
             // indicate acceptance either way at this point.
-            _ = parseArgumentsText(from: line)
             endLocation = line.location!
```
Comment (franklinsch) [inline Sources/Markdown/Parser/BlockDirectiveParser.swift:104]: Do we still need this line? If `endLocation` is set in `parseArgumentsText(from:)`, won't that override the value set here?

Comment (franklinsch) [inline Sources/Markdown/Parser/BlockDirectiveParser.swift:104]: Do we still need this line? If `endLocation` is set in `parseArgumentsText(from:)`, won't that override the value set here?

Comment (Kyle-Ye) [reply Sources/Markdown/Parser/BlockDirectiveParser.swift:104]: Yes, it will be overridden later anyway. I'll just remove it.

**Comment by Kyle-Ye** [comment]
@swift-ci please test

**Review by franklinsch — APPROVED** [review]

**Pull request merged**